### ISSUE TITLE
Implement more Components

### ISSUE THREAD

**Issue by dogamak** [dogamak/xcbars#3]
- [x] Pipe that refreshes every X seconds
- [x] Current window title
- [ ] Smart clock component that figures out the needed refresh frequency
- [ ] Workspaces
- [ ] Fifo
- [ ] Netcat / RPC Component that shows text sent to it via TCP
- [ ] Wifi

**Comment by dogamak** [comment]
1f0402df826f65f53c288858035b1216c94f26bd implements the first bullet point.

**Comment by chrisduerr** [comment]
With stuff like Workspaces a WM/DE specific solution is required as far as I know. Implementing such non-trivial components directly into the library could lead to quite a lot of stuff being unused for most bars. What do you think about this "issue", if it is an issue at all?

Assuming we have a billion components in a week, it could make the project a bit of a mess.

**Comment by dogamak** [comment]
Yes, WM specific stuff does not belong to this crate, however I think the workspace component can be implemented for all browsers that support EWHM. EWHM has [standardized properties for workspace information](https://specifications.freedesktop.org/wm-spec/wm-spec-latest.html#idm140200472709456), such as the workspace names and the currently active workspace. With supporting these we can support all EWHM compliant window manages and desktop environments, [which are quite many](https://en.wikipedia.org/wiki/Extended_Window_Manager_Hints#List_of_window_managers_that_support_Extended_Window_Manager_Hints).

**Comment by chrisduerr** [comment]
Oh I didn't know about that. This would definitely be neat!

**Comment by chrisduerr** [comment]
I'm working on the current window title component right now.

**Comment by chrisduerr** [comment]
I'm also gonna get into the workspaces thing. Should be pretty similar to the window title.

**Comment by chrisduerr** [comment]
@dogamak Merge #36 and then tick off the Window Title?

**Comment by dogamak** [comment]
Ah, yes, sorry.
